src/solvers/flattening/boolbv_array_of.cpp

```diff
@@ -45,8 +46,10 @@ bvt boolbvt::convert_array_of(const array_of_exprt &expr)
   bvt bv;
   bv.resize(width);
 
-  if(size*tmp.size()!=width)
-    throw "convert_array_of: unexpected operand width";
+  INVARIANT(
+    size * tmp.size() == width,
```

Comment:
This should be moved further up, as soon as `size`, `tmp`, and `width` are known.